src/metabase/models/params.clj

```diff
@@ -59,10 +61,11 @@
   [param-field-ids]
   (not-empty
    (into {}
-         (map (comp (juxt :field_id identity)
-                    #(select-keys % [:field_id :human_readable_values :values])
-                    #(field-values/get-latest-full-field-values (:id %))))
-         (t2/hydrate (t2/select :model/Field :id [:in (set param-field-ids)]) :values))))
+         (comp (map field-values/get-latest-full-field-values)
+               (remove nil?)
+               (map #(select-keys % [:field_id :human_readable_values :values]))
+               (map (juxt :field_id identity)))
+         param-field-ids)))
```

Comment:
This is rather unrelated fix. I've noticed existing implementation returns map containing `[nil {}]` entry that is a result of processing fields that have no field values. Also db call is removed as it is not necessary. If I recall correctly it was not before the use of `field-values/get-latest-full-field-values`. But this function does db call for every field id instead of checking hydrated values so now it is.